### ISSUE TITLE
Remote-SSH causes Azure VPS ssh exception 

### ISSUE THREAD

**Issue by SlyLi** [microsoft/vscode-remote-release#5211]
Issue Type: <b>Bug</b>

Azure VPS ：Linux (ubuntu 20.04)
After remote-ssh uses the key to connect to vps for a period of time, there is a problem that vps needs to be reconnected, and then other ssh software is used to connect and cannot be connected. After restarting vps, ssh returns to normal. This problem still occurs when using remote-ssh. But there is no problem when only connecting with other ssh software.

Extension version: 0.65.4
VS Code version: Code 1.55.2 (3c4e3df9e89829dce27b7b5c24508306b151f30d, 2021-04-13T09:35:57.887Z)
OS version: Windows_NT x64 10.0.19041

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz (4 x 2304)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.89GB (8.34GB free)|
|Process Argv|--crash-reporter-id e72a10be-0b82-4ba1-b936-58e8ffa9fef0|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383cf:30185419
pythonvspyt602:30300191
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492cf:30256860
vstes627:30244334
pythonvspyt639:30300192
pythontb:30283811
pythonvspyt551cf:30311713
vspre833:30321513
pythonptprofiler:30281270
vshan820:30294714
pythondataviewer:30285071
vscus158:30321503
vscorehov:30309549
vscod805:30301674
binariesv615cf:30321889
vsccppwtct:30322305

```

</details>

**Comment by bamurtaugh** [comment]
Do you have any sort of SSH output logs you can share when you get disconnected? https://code.visualstudio.com/docs/remote/troubleshooting#_reporting-issues

**Comment by SlyLi** [comment]
> Do you have any sort of SSH output logs you can share when you get disconnected? https://code.visualstudio.com/docs/remote/troubleshooting#_reporting-issues


<details>
<summary>SSH output logs </summary>

```
[16:56:44.050] Log Level: 2
[16:56:44.057] remote-ssh@0.65.4
[16:56:44.057] win32 x64
[16:56:44.059] SSH Resolver called for "ssh-remote+xx.xx.xx.xx", attempt 1
[16:56:44.059] "remote.SSH.useLocalServer": false
[16:56:44.060] "remote.SSH.showLoginTerminal": false
[16:56:44.060] "remote.SSH.remotePlatform": {"8.131.57.193":"linux","xx.xx.xx.xx":"linux","xx.xx.xx.xx":"linux","xx.xx.xx.xx":"linux","xx.xx.xx.xx":"linux","xx.xx.xx.xx":"linux"}
[16:56:44.060] "remote.SSH.path": undefined
[16:56:44.060] "remote.SSH.configFile": undefined
[16:56:44.061] "remote.SSH.useFlock": true
[16:56:44.061] "remote.SSH.lockfilesInTmp": false
[16:56:44.061] "remote.SSH.localServerDownload": auto
[16:56:44.061] "remote.SSH.remoteServerListenOnSocket": false
[16:56:44.061] "remote.SSH.showLoginTerminal": false
[16:56:44.061] "remote.SSH.defaultExtensions": []
[16:56:44.062] "remote.SSH.loglevel": 2
[16:56:44.062] SSH Resolver called for host: xx.xx.xx.xx
[16:56:44.062] Setting up SSH remote "xx.xx.xx.xx"
[16:56:44.090] Using commit id "3c4e3df9e89829dce27b7b5c24508306b151f30d" and quality "stable" for server
[16:56:44.096] Install and start server if needed
[16:56:44.128] Checking ssh with "ssh -V"
[16:56:44.187] > OpenSSH_for_
[16:56:44.188] > Windows_8.1p1, LibreSSL 3.0.2

[16:56:44.195] Running script with connection command: ssh -T -D 8700 "xx.xx.xx.xx" bash
[16:56:44.199] Terminal shell path: C:\Windows\System32\cmd.exe
[16:56:45.293] >  ]0;C:\Windows\System32\cmd.exe
[16:56:45.293] Got some output, clearing connection timeout
[16:56:48.455] > 0127cd291377: running
[16:56:48.502] > Acquiring lock on /home/Joseph/.vscode-server/bin/3c4e3df9e89829dce27b7b5c245083
> 06b151f30d/vscode-remote-lock.Joseph.3c4e3df9e89829dce27b7b5c24508306b151f30d   
[16:56:48.520] > Found existing installation at /home/Joseph/.vscode-server/bin/3c4e3df9e89829dce
> 27b7b5c24508306b151f30d...
> Checking /home/Joseph/.vscode-server/.3c4e3df9e89829dce27b7b5c24508306b151f30d.l
> og and /home/Joseph/.vscode-server/.3c4e3df9e89829dce27b7b5c24508306b151f30d.pid
>  for a running server
> Looking for server with pid: 3828
[16:56:48.565] > Starting server with command... /home/Joseph/.vscode-server/bin/3c4e3df9e89829dc
> e27b7b5c24508306b151f30d/server.sh --start-server --host=127.0.0.1 --enable-remo
> te-auto-shutdown  --port=0  &> "/home/Joseph/.vscode-server/.3c4e3df9e89829dce27
> b7b5c24508306b151f30d.log" < /dev/null
> printenv:
>     XDG_SESSION_ID=3
>     SELINUX_ROLE_REQUESTED=
>     SHELL=/bin/bash
>     SSH_CLIENT=101.76.240.166 8703 22
>     SELINUX_USE_CURRENT_RANGE=
>     USER=Joseph
>     VSCODE_AGENT_FOLDER=/home/Joseph/.vscode-server
>     PATH=/usr/local/bin:/usr/bin
>     MAIL=/var/mail/Joseph
>     PWD=/home/Joseph
>     LANG=en_US.UTF-8
>     SELINUX_LEVEL_REQUESTED=
>     HOME=/home/Joseph
>     SHLVL=2
>     LOGNAME=Joseph
>     SSH_CONNECTION=101.76.240.166 8703 10.1.1.4 22
>     LESSOPEN=||/usr/bin/lesspipe.sh %s
>     XDG_RUNTIME_DIR=/run/user/1000
>     _=/usr/bin/printenv
[16:56:48.572] > 
> Spawned remote server: 1659
> Waiting for server log...
[16:56:48.580] > 
[16:56:48.615] > Waiting for server log...
[16:56:48.656] > Waiting for server log...
[16:56:48.698] > Waiting for server log...
[16:56:48.726] > Waiting for server log...
[16:56:48.736] > 
[16:56:48.767] > Waiting for server log...
[16:56:48.808] > Waiting for server log...
[16:56:48.836] > Waiting for server log...
[16:56:48.885] > Waiting for server log...
[16:56:48.913] > Waiting for server log...
[16:56:48.953] > Waiting for server log...
[16:56:48.987] > Waiting for server log...
[16:56:49.025] > Waiting for server log...
[16:56:49.062] > Waiting for server log...
[16:56:49.098] > Waiting for server log...
[16:56:49.141] > Waiting for server log...
[16:56:49.174] > Waiting for server log...
[16:56:49.215] > Waiting for server log...
[16:56:49.248] > Waiting for server log...
[16:56:49.290] > Waiting for server log...
[16:56:49.383] >  
> *
> * Reminder: You may only use this software with Visual Studio family products,  
> * as described in the license (https://go.microsoft.com/fwlink/?linkid=2077057) 
> *
> 
[16:56:49.449] > 0127cd291377: start
> SSH_AUTH_SOCK====
> DISPLAY====
> webUiAccessToken====
> listeningOn==34308==
> osReleaseId==centos==
> arch==x86_64==
> tmpDir==/run/user/1000==
> platform==linux==
> unpackResult====
> didLocalDownload==0==
> downloadTime====
> installTime====
> extInstallTime====
> serverStartTime==810==
> connectionToken==1aa11111-11aa-1111-aa11-aa1aa11a1a1a==
> 0127cd291377: end
[16:56:49.450] Received install output: 
SSH_AUTH_SOCK====
DISPLAY====
webUiAccessToken====
listeningOn==34308==
osReleaseId==centos==
arch==x86_64==
tmpDir==/run/user/1000==
platform==linux==
unpackResult====
didLocalDownload==0==
downloadTime====
installTime====
extInstallTime====
serverStartTime==810==
connectionToken==1aa11111-11aa-1111-aa11-aa1aa11a1a1a==

[16:56:49.451] Remote server is listening on 34308
[16:56:49.451] Parsed server configuration: {"serverConfiguration":{"remoteListeningOn":{"port":34308},"osReleaseId":"centos","arch":"x86_64","webUiAccessToken":"","sshAuthSock":"","display":"","tmpDir":"/run/user/1000","platform":"linux","connectionToken":"1aa11111-11aa-1111-aa11-aa1aa11a1a1a"},"serverStartTime":810,"installUnpackCode":""}
[16:56:49.453] Starting forwarding server. localPort 8720 -> socksPort 8700 -> remotePort 34308
[16:56:49.455] Forwarding server listening on 8720
[16:56:49.455] Waiting for ssh tunnel to be ready
[16:56:49.457] Tunneled 34308 to local port 8720
[16:56:49.457] Resolved "ssh-remote+xx.xx.xx.xx" to "127.0.0.1:8720"
[16:56:49.459] [Forwarding server 8720] Got connection 0
[16:56:49.469] ------



[16:56:49.523] [Forwarding server 8720] Got connection 1
[16:56:49.938] [Forwarding server 8720] Got connection 2
[17:29:01.447] ------



[17:29:01.447] SSH Resolver called for "ssh-remote+xx.xx.xx.xx", attempt 2, (Reconnection)
[17:29:01.447] SSH Resolver called for host: xx.xx.xx.xx
[17:29:01.447] Setting up SSH remote "xx.xx.xx.xx"
[17:29:01.452] Using commit id "3c4e3df9e89829dce27b7b5c24508306b151f30d" and quality "stable" for server
[17:29:01.457] Install and start server if needed
[17:29:01.464] Running script with connection command: ssh -T -D 5786 "xx.xx.xx.xx" bash
[17:29:01.466] Terminal shell path: C:\Windows\System32\cmd.exe
[17:29:01.583] >  ]0;C:\Windows\System32\cmd.exe
[17:29:01.584] Got some output, clearing connection timeout
[17:29:17.845] > c3a0ccee99dd: running
[17:29:17.896] > Acquiring lock on /home/Joseph/.vscode-server/bin/3c4e3df9e89829dce27b7b5c245083
> 06b151f30d/vscode-remote-lock.Joseph.3c4e3df9e89829dce27b7b5c24508306b151f30d   
[17:29:17.915] > Found existing installation at /home/Joseph/.vscode-server/bin/3c4e3df9e89829dce
> 27b7b5c24508306b151f30d...
> Checking /home/Joseph/.vscode-server/.3c4e3df9e89829dce27b7b5c24508306b151f30d.l
> og and /home/Joseph/.vscode-server/.3c4e3df9e89829dce27b7b5c24508306b151f30d.pid
>  for a running server
> Looking for server with pid: 1659
[17:29:17.937] > Found running server...
[17:29:17.947] > 
> *
> * Reminder: You may only use this software with Visual Studio family products, 
> * as described in the license (https://go.microsoft.com/fwlink/?linkid=2077057)
> *
> 
[17:29:17.961] > Checking server status on port 34308 with wget
[17:29:18.187] > c3a0ccee99dd: start  
> SSH_AUTH_SOCK====    
> DISPLAY====
> webUiAccessToken==== 
> listeningOn==34308== 
> osReleaseId==centos==
> arch==x86_64==       
> tmpDir==/run/user/1000==
> platform==linux==
> unpackResult====
> didLocalDownload==0==
> downloadTime====
> installTime====
> extInstallTime====
> serverStartTime====
> connectionToken==1aa11111-11aa-1111-aa11-aa1aa11a1a1a==
> c3a0ccee99dd: end
[17:29:18.188] Received install output:   
SSH_AUTH_SOCK====    
DISPLAY====
webUiAccessToken==== 
listeningOn==34308== 
osReleaseId==centos==
arch==x86_64==       
tmpDir==/run/user/1000==
platform==linux==
unpackResult====
didLocalDownload==0==
downloadTime====
installTime====
extInstallTime====
serverStartTime====
connectionToken==1aa11111-11aa-1111-aa11-aa1aa11a1a1a==

[17:29:18.188] Remote server is listening on 34308
[17:29:18.189] Parsed server configuration: {"serverConfiguration":{"remoteListeningOn":{"port":34308},"osReleaseId":"centos","arch":"x86_64","webUiAccessToken":"","sshAuthSock":"","display":"","tmpDir":"/run/user/1000","platform":"linux","connectionToken":"1aa11111-11aa-1111-aa11-aa1aa11a1a1a"},"installUnpackCode":""}
[17:29:18.191] Starting forwarding server. localPort 11726 -> socksPort 5786 -> remotePort 34308
[17:29:18.192] Forwarding server listening on 11726
[17:29:18.193] Waiting for ssh tunnel to be ready
[17:29:18.195] Tunneled 34308 to local port 11726
[17:29:18.195] Resolved "ssh-remote+xx.xx.xx.xx" to "127.0.0.1:11726"
[17:29:18.196] [Forwarding server 11726] Got connection 0
[17:29:18.218] ------



[17:29:18.295] [Forwarding server 11726] Got connection 1
[17:29:18.719] [Forwarding server 11726] Got connection 2
[17:30:19.162] [Forwarding server 11726] Got connection 3
[17:30:19.522] ------



[17:30:19.522] SSH Resolver called for "ssh-remote+xx.xx.xx.xx", attempt 3, (Reconnection)
[17:30:19.522] SSH Resolver called for host: xx.xx.xx.xx
[17:30:19.523] Setting up SSH remote "xx.xx.xx.xx"
[17:30:19.526] Using commit id "3c4e3df9e89829dce27b7b5c24508306b151f30d" and quality "stable" for server
[17:30:19.530] Install and start server if needed
[17:30:19.536] Running script with connection command: ssh -T -D 8087 "xx.xx.xx.xx" bash
[17:30:19.538] Terminal shell path: C:\Windows\System32\cmd.exe
[17:30:19.636] >  ]0;C:\Windows\System32\cmd.exe
[17:30:19.636] Got some output, clearing connection timeout
[17:30:34.762] > 0877befdcf6e: running
[17:30:34.993] > Acquiring lock on /home/Joseph/.vscode-server/bin/3c4e3df9e89829dce27b7b5c245083
> 06b151f30d/vscode-remote-lock.Joseph.3c4e3df9e89829dce27b7b5c24508306b151f30d   
[17:30:35.063] > Found existing installation at /home/Joseph/.vscode-server/bin/3c4e3df9e89829dce
> 27b7b5c24508306b151f30d...
> Checking /home/Joseph/.vscode-server/.3c4e3df9e89829dce27b7b5c24508306b151f30d.l
> og and /home/Joseph/.vscode-server/.3c4e3df9e89829dce27b7b5c24508306b151f30d.pid
>  for a running server
[17:30:35.087] > Looking for server with pid: 1659
[17:30:35.118] > Found running server...
[17:30:35.135] > 
> *
> * Reminder: You may only use this software with Visual Studio family products, 
> * as described in the license (https://go.microsoft.com/fwlink/?linkid=2077057)
> *
> 
[17:30:35.145] > Checking server status on port 34308 with wget
[17:30:35.545] > 0877befdcf6e: start  
> SSH_AUTH_SOCK====    
> DISPLAY====
> webUiAccessToken==== 
> listeningOn==34308== 
> osReleaseId==centos==
> arch==x86_64==       
> tmpDir==/run/user/1000==
> platform==linux==
> unpackResult====
> didLocalDownload==0==
> downloadTime====
> installTime====
> extInstallTime====
> serverStartTime====
> connectionToken==1aa11111-11aa-1111-aa11-aa1aa11a1a1a==
> 0877befdcf6e: end
[17:30:35.545] Received install output:   
SSH_AUTH_SOCK====    
DISPLAY====
webUiAccessToken==== 
listeningOn==34308== 
osReleaseId==centos==
arch==x86_64==       
tmpDir==/run/user/1000==
platform==linux==
unpackResult====
didLocalDownload==0==
downloadTime====
installTime====
extInstallTime====
serverStartTime====
connectionToken==1aa11111-11aa-1111-aa11-aa1aa11a1a1a==

[17:30:35.546] Remote server is listening on 34308
[17:30:35.547] Parsed server configuration: {"serverConfiguration":{"remoteListeningOn":{"port":34308},"osReleaseId":"centos","arch":"x86_64","webUiAccessToken":"","sshAuthSock":"","display":"","tmpDir":"/run/user/1000","platform":"linux","connectionToken":"1aa11111-11aa-1111-aa11-aa1aa11a1a1a"},"installUnpackCode":""}
[17:30:35.551] Starting forwarding server. localPort 8106 -> socksPort 8087 -> remotePort 34308
[17:30:35.556] Forwarding server listening on 8106
[17:30:35.556] Waiting for ssh tunnel to be ready
[17:30:35.559] Tunneled 34308 to local port 8106
[17:30:35.560] Resolved "ssh-remote+xx.xx.xx.xx" to "127.0.0.1:8106"
[17:30:35.561] [Forwarding server 8106] Got connection 0
[17:30:35.568] ------



[17:30:35.606] [Forwarding server 8106] Got connection 1
[17:30:49.657] [Forwarding server 8106] Got connection 2



```

</details>

That's my local logs and  I replaced my IPs to xx.xx.xx.xx

I have changed my VPS system to centos, but this disconnection still happends. I only found this kinds of error when I edit larg javascript(.js) file, and the VSCODE's notice sub window keep showing `Initializing JS/TS language features `

**Comment by bamurtaugh** [comment]
Does deactivating the built-in "TypeScript and JavaScript Language Features" extension make any difference?

**Comment by SlyLi** [comment]
> Does disabling the built-in "TypeScript and JavaScript Language Features" extension make any difference?

It works.  Thanks .

**Comment by bamurtaugh** [comment]
cc @mjbvz for any thoughts on the JS/TS language features initialization; I've seen a few closed out issues related to this (https://github.com/microsoft/vscode-remote-release/issues/3298, https://github.com/microsoft/vscode-remote-release/issues/3173, https://github.com/microsoft/vscode-remote-release/issues/383) but not sure if there's anything else we should do here

**Comment by SlyLi** [comment]
> cc @mjbvz for any thoughts on the JS/TS language features initialization; I've seen a few closed out issues related to this (#3298, #3173, #383) but not sure if there's anything else we should do here

When I open large file (just log file, 1-5M ), the disconnection still happends , I wonder if it is possible there are something wrong while sending file data, split data or merge data

**Comment by roblourens** [comment]
Extensions working on large files can cause these disconnections, tracking those issues here https://github.com/microsoft/vscode/issues/135021
